public/app/percona/shared/components/hooks/recurringCall.hook.ts

```diff
@@ -0,0 +1,37 @@
+import { logger } from '@percona/platform-core';
+import { useRef, useEffect } from 'react';
+
+type hookRecurringCallback = () => void;
+
+export const useRecurringCall = () => {
+  const timer = useRef<number>();
+  const interval = useRef<number>();
+
+  const triggerTimeout = async (cb: hookRecurringCallback, defaultInterval = 10000, callImmediate = false) => {
+    interval.current = defaultInterval;
+    try {
+      callImmediate && (await cb());
+      timer.current = window.setTimeout(async () => {
```

Comment:
```suggestion
      timer.current = setTimeout(async () => {
```

Reply:
If we remove `window`, TypeScript will assume we're dealing with NodeJS' `setTimeout`, which returns a `NodeJS.Timeout` object. In the browser however, `setTimeout` returns a number ID instead, so we need to explicitly add `window` to the call and make it unambiguous.

Reply:
I don't see what's the problem with `NodeJS.Timeout`. Also: https://stackoverflow.com/a/56239226

Reply:
Wasn't aware of this `ReturnType`, nice one !